modules/stereo/src/descriptor.hpp

```diff
@@ -0,0 +1,451 @@
+                                {
+                                    int rrWidth = ii * width;
+                                    for (int jj = j - step; jj <= j + step; jj += step)
+                                    {
+                                        if (image[d][rrWidth + jj] > image[d][rWidth + j])
+                                        {
+                                            c[d] = c[d] + 1;
+                                        }
+                                        c[d] = c[d] * 2;
+                                    }
+                                }
+                            }
+                            for (int ii = -1; ii <= +1; ii++)
+                            {
+                                int rrWidth = (ii + i) * width;
+
+                                if (i == -1)
+                                {
+                                    if (ii + i != i)
+                                    {
+                                        if (image[d][rrWidth + j] > image[d][rWidth + j])
+                                        {
+                                            c[d] = c[d] + 1;
+                                        }
+                                        c[d] = c[d] * 2;
+                                    }
+                                }
+                                else if (i == 0)
+                                {
+                                    for (int j2 = -1; j2 <= 1; j2 += 2)
+                                    {
+                                        if (ii + i != i)
+                                        {
+                                            if (image[d][rrWidth + j + j2] > image[d][rWidth + j])
+                                            {
+                                                c[d] = c[d] + 1;
+                                            }
+                                            c[d] = c[d] * 2;
+                                        }
+                                    }
+                                }
+                                else
+                                {
+                                    if (ii + i != i)
+                                    {
+                                        if (image[d][rrWidth + j] > image[d][rWidth + j])
+                                        {
+                                            c[d] = c[d] + 1;
+                                        }
+                                        c[d] = c[d] * 2;
+                                    }
+                                }
+                            }
+                            dst[d][rWidth + j] = c[d];
+                            dst[d][rWidth + j] = c[d];
+                        }
+                    }
+                }
+            }
+        };
+        //!paralel implementation of the center symetric census
+        class parallelSymetricCensus:public ParallelLoopBody
```

Comment:
`parallel` in the name is misleading. Please, make those descriptors work on 1 or 2 images with compile-time parameter.